services/src/main/java/org/exoplatform/task/rest/TaskRestService.java

```diff
@@ -491,8 +507,16 @@ public Response getLabels() {
       @ApiResponse(code = 500, message = "Internal server error") })
   public Response getLabelsByTaskId(@ApiParam(value = "Task id", required = true) @PathParam("id") long id) {
     try {
-    String currentUser = ConversationState.getCurrent().getIdentity().getUserId();
-    return Response.ok(labelService.findLabelsByTask(id, currentUser, 0, -1)).build();
+    Identity currentUser = ConversationState.getCurrent().getIdentity();
```

Comment:
@mkrout  do we have a permission check here ? A user with no access to the task should not be able to retrieve its labels.